site/_posts/2021-03-16-securely-implement-oauth-vuejs.md

```diff
@@ -1174,32 +1221,310 @@ methods: {
+
+> <strong> Authentication is <u>who</u> you are. </strong>
+
+> <strong> Authorization is <u>what</u> you can do. </strong>  
+
+We will add feedback when a user is authenticated, but not authorized to update user data or access our application.  Please note in some cases, you may not want to give such an acknowledgement, but rather simply route the user back to a login screen with a message of "Login Failed."  
+
+We will do this by adding a new data property called `authState` to our server response with the following possible values: 
+- `Authorized`
+- `notAuthorized`
+- `notAuthenticated`
+
+## Update the server code
+
+We will start by adding to our `user.js` axios request. 
+
+```javascript
+router.get("/", (req, res) => {
+  // token in session -> get user data and send it back to the vue app
+  if (req.session.token) {
+    axios
+      .post(
+        `http://localhost:${process.env.FUSIONAUTH_PORT}/oauth2/introspect`,
+        qs.stringify({
+          client_id: process.env.CLIENT_ID,
+          token: req.session.token,
+        })
+      )
+      .then((result) => {
+        let introspectResponse = result.data;
+        // valid token -> get more user data and send it back to the Vue app
+        if (introspectResponse) {
+
+          // GET request to /registration endpoint
+          axios
+            .get(
+              `http://localhost:${process.env.FUSIONAUTH_PORT}/api/user/registration/${introspectResponse.sub}/${process.env.APPLICATION_ID}`, {
+                headers: {
+                  Authorization: process.env.API_KEY,
+                },
+              }
+            )
+            .then((response) => {
+              res.send({
+                authState: "Authorized",
+                introspectResponse: introspectResponse,
+                body: response.data.registration,
+              });
+            })
+            .catch((err) => {
+              res.send({
+                authState: "notAuthorized"
+              });
+              console.log(err)
+              return
+            })
+        }
+        // expired token -> send nothing
+        else {
+          req.session.destroy();
+          res.send({
+            authState: "notAuthenticated"
+          });
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+  // no token -> send nothing
+  else {
+    res.send({
+      authState: "notAuthenticated"
+    });
+  }
+});
+```
+
+## Update the client code
+#### Add to your `App.vue`
+
+We will pass additional props to our `Greet` and `Login` components.  Additionally, we have added `boolShowSignOut`, rather than rendering only based on `email` to better indicate if the user should be presented with a `sign in` or `sign out` button.  
```

Comment:
Is having a type in the var name (`boolShowSignout`) common in Vue code? If at all possible, let's keep var names in line with the Vue community for this post.

Reply:
Good point, based on what I was reading it would seem that something like `showSignOut` might be more appropriate.